central/config/datastore/datastore_test.go

```diff
@@ -203,3 +208,274 @@ func (s *configDataStoreTestSuite) TestAllowsUpdate() {
+			initialConfig: &storage.Config{
+				PublicConfig: samplePublicConfig,
+				PrivateConfig: &storage.PrivateConfig{
+					AlertRetention:                      customAlertRetention,
+					ImageRetentionDurationDays:          DefaultImageRetention + 1,
+					ExpiredVulnReqRetentionDurationDays: DefaultExpiredVulnReqRetention + 1,
+					DecommissionedClusterRetention:      nil,
+					ReportRetentionConfig:               customReportRetentionConfig,
+					VulnerabilityExceptionConfig:        customVulnerabilityDeferralConfig,
+					AdministrationEventsConfig:          customAdministrationEventsConfig,
+				},
+			},
+			upsertedConfig: &storage.Config{
+				PublicConfig: samplePublicConfig,
+				PrivateConfig: &storage.PrivateConfig{
+					AlertRetention:                      customAlertRetention,
+					ImageRetentionDurationDays:          DefaultImageRetention + 1,
+					ExpiredVulnReqRetentionDurationDays: DefaultExpiredVulnReqRetention + 1,
+					DecommissionedClusterRetention:      defaultDecommissionedClusterRetention,
+					ReportRetentionConfig:               customReportRetentionConfig,
+					VulnerabilityExceptionConfig:        customVulnerabilityDeferralConfig,
+					AdministrationEventsConfig:          customAdministrationEventsConfig,
+				},
+			},
+		},
+		"Configure report retention when missing": {
+			initialConfig: &storage.Config{
+				PublicConfig: samplePublicConfig,
+				PrivateConfig: &storage.PrivateConfig{
+					AlertRetention:                      customAlertRetention,
+					ImageRetentionDurationDays:          DefaultImageRetention + 1,
+					ExpiredVulnReqRetentionDurationDays: DefaultExpiredVulnReqRetention + 1,
+					DecommissionedClusterRetention:      customDecommissionedClusterRetention,
+					ReportRetentionConfig:               nil,
+					VulnerabilityExceptionConfig:        customVulnerabilityDeferralConfig,
+					AdministrationEventsConfig:          customAdministrationEventsConfig,
+				},
+			},
+			upsertedConfig: &storage.Config{
+				PublicConfig: samplePublicConfig,
+				PrivateConfig: &storage.PrivateConfig{
+					AlertRetention:                      customAlertRetention,
+					ImageRetentionDurationDays:          DefaultImageRetention + 1,
+					ExpiredVulnReqRetentionDurationDays: DefaultExpiredVulnReqRetention + 1,
+					DecommissionedClusterRetention:      customDecommissionedClusterRetention,
+					ReportRetentionConfig:               defaultReportRetentionConfig,
+					VulnerabilityExceptionConfig:        customVulnerabilityDeferralConfig,
+					AdministrationEventsConfig:          customAdministrationEventsConfig,
+				},
+			},
+		},
+		"Configure vulnerability exception management when missing and Feature activated": {
+			enabledFlags: []string{features.UnifiedCVEDeferral.EnvVar()},
+			initialConfig: &storage.Config{
+				PublicConfig: samplePublicConfig,
+				PrivateConfig: &storage.PrivateConfig{
+					AlertRetention:                      customAlertRetention,
```

Comment:
Do you think we should `clone` fields to ensure we are not comparing pointers but values?